PyDSS/pyPostprocessor/PostprocessScripts/AutomatedThermalUpgrade.py

```diff
@@ -186,21 +183,61 @@ def __init__(self, project, scenario, inputs, dssInstance, dssSolver, dssObjects
 
         # Mitigate thermal violations
         self.Line_trial_counter = 0
-        while len(self.xfmr_violations) > 0 or len(self.line_violations) > 0:
+        # TODO change to 20
+        self.max_upgrade_iteration = min(5, len(self.xfmr_violations) + len(self.line_violations))
+        while (len(self.xfmr_violations) > 0 or len(self.line_violations) > 0) \
+                and (self.Line_trial_counter < self.max_upgrade_iteration):

```

Comment:
This `while` condition causes infinite loops with some edge cases, have you updated the logics in it to make sure the condition could be met and exit at some point? or you can add a timedelta to control the timeout as we discussed?

Reply:
Added that from another branch now:
-limit on number of parallel transformers possible
- time limit exception function